/*
 * Copyright (C) 2020 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package dumpstate_proto;

import "google/protobuf/empty.proto";

message ServiceNameList {
    repeated string service_names = 1;
}

message ServiceLogRequest {
    string service_name = 1;
}

message DumpstateBuffer {
    bytes buffer = 1;
}

service DumpstateServer {
    rpc GetSystemLogs(google.protobuf.Empty) returns (stream DumpstateBuffer) {}

    rpc GetAvailableServices(google.protobuf.Empty) returns (ServiceNameList) {}

    rpc GetServiceLogs(ServiceLogRequest) returns (stream DumpstateBuffer) {}
}
